test-butler-app-physical-devices/src/main/java/com/linkedin/android/testbutler/shell/ShellLocationModeSetting.java

```diff
@@ -161,20 +161,23 @@ private boolean clickNetworkConsentAgreeButton() {
 
             Log.d(TAG, "Network provider was disabled, looking for consent dialog");
 
-            boolean clickedAgree = false;
-            for (int i = 0; i < NETWORK_CONTENT_CLICK_ATTEMPTS; i++) {
-                clickedAgree = clickedAgree || tryClickingConsentAgreeButton(i);
-
-                if (clickedAgree) {
-                    if (isProviderEnabled(getLocationProviders(), LocationManager.NETWORK_PROVIDER)) {
-                        return true;
+            try (UiAutomationConnectionWrapper uiAutomation = UiAutomationConnectionWrapper.newInstance()) {
```

Comment:
Does this actually work? The UIAutomationConnectionWRapper implements Closeable, not AutoCloseabble (which i don't think we can use since that was introduced  in 19 and we min 14 atm). But I don't think Closeable is  compatible with try-with-resources,  I think you need  AutoCloseable for that (please correct me if I'm wrong)

Reply:
The try-with-resources is just moved from elsewhere, so it definitely works. Closeable extends AutoCloseable, not the other way around. I guess for <19 they just support Closeable. https://developer.android.com/studio/write/java8-support.html#supported_features - says they support try-with-resources at all language levels.

Reply:
ah cool, thanks for the correction :)